handlers.go

```diff
@@ -62,3 +62,22 @@ func RecoverImmediately(ctx context.Context, dagst *DAGStore, failureCh chan Sha
 		}
 	}
 }
+
+// LogTraceLoop logs the output of the given trace channel until the given context expires.
+func LogTraceLoop(ctx context.Context, traceCh chan Trace, onDone func()) {
```

Comment:
This doesn't belong here. As a utility, this makes way too many assumptions: log level, message, etc. The leakage is obvious in the fact that you need to provide an `onDone` function as an argument.

Reply:
Removed this.